fairscale/optim/oss.py

```diff
@@ -555,20 +547,34 @@ def _broadcast_params(self) -> None:
         # Exchange all the shards with the other ranks
         for device in self.buckets.keys():
             for dst_rank, bucket in self.buckets[device].items():
-                work_handles.append(
-                    dist.broadcast(
-                        tensor=bucket.buffer, src=self._local_to_global_rank[dst_rank], group=self.group, async_op=True,
+                self._work_handles.append(
+                    Workhandle(
+                        dist.broadcast(
+                            tensor=bucket.buffer,
+                            src=self._local_to_global_rank[dst_rank],
+                            group=self.group,
+                            async_op=True,
+                        ),
+                        None,
                     )
                 )
 
-        _ = list(filter(lambda x: x.wait(), work_handles))
+        self._consume_work_handles()
```

Comment:
I know that I went back and forth on that, but exposing these handles now so that ShardedDPP can use them in a subsequent PR. The idea is that ShardedDDP can fire all the reduce calls, but does not have to block for all of them to be done before handing it over to OSS, it just has to wait for the gradients this rank owns. It can then pass the handles to OSS, which has to wait for them to be done + subsequent broadcasts before the next FW. This way some of the step+broadcast could overlap with the BW